sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRelation.scala

```diff
@@ -96,7 +97,16 @@ private[sql] case class JDBCRelation(
 
   override val needConversion: Boolean = false
 
-  override val schema: StructType = JDBCRDD.resolveTable(url, table, properties)
+  override val schema: StructType = {
+    val resolvedSchema = JDBCRDD.resolveTable(url, table, properties)
+    providedSchemaOption match {
+      case Some(providedSchema) =>
+        if (providedSchema.sql.toLowerCase == resolvedSchema.sql.toLowerCase) resolvedSchema
```

Comment:
I guess it would make sense if it does not try to apply the resolved schema when the schema is explicitly set like the other data sources.